internal/mocha_node_test/mocha_node_test.bzl

```diff
@@ -0,0 +1,50 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""NodeJS testing
+
+These rules let you run tests using the Mocha library.
+
+Unlike the typical mocha testing setup, we do not use the mocha binary that is
+provided by the mocha library. The method used by these rules to patch the
+module loader with extra module name -> path mappings (node_loader.js) does not
+persist across processes. The mocha binary provided by the mocha library
+ultimately spawns a new process to run the actual tests, and so the patched
+module loader will not be respected. This approach instead programmatically runs
+the Mocha library on the desired files; see
+https://github.com/mochajs/mocha/wiki/Using-mocha-programmatically.
+"""
+load("//internal/node:node.bzl", "nodejs_test")
+
+def mocha_node_test(
+  name,
+  test_entrypoints = [], # these should NOT contain the preceding workspace name.
+  data = [],
+  expected_exit_code = 0,
+  **kwargs):
+
```

Comment:
Needs a docstring here for this function. See the jasmine_node_test.bzl for an example. This will fix the lint failure: `./internal/mocha_node_test/mocha_node_test.bzl:30:1-37:2: function 'mocha_node_test' has no docstring`